contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol

```diff
@@ -159,7 +159,9 @@ contract OVM_ExecutionManager is iOVM_ExecutionManager, Lib_AddressResolver {
         override
         public
     {
-        require(transactionContext.ovmNUMBER == 0, "Only be callable at the start of a transaction");
+        require(transactionContext.ovmNUMBER == 0, "Only callable at the start of a transaction");
+        require(_transaction.blockNumber != 0, "Prevent reentrancy by disallowing default context values");
```

Comment:
Agree that this is a fine way to go, although we need to make sure this is handled the same way as an invalid gas limit in `run`.  This is less safe because if you can ever satisfy this require's condition, then you could make proving fraud impossible, since reverts are reserved for `INVALID_STATE_ACCESS`.  Should probably break this and the gaslimit one into a shared internal function.